src/client/index.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable no-console */
+/* eslint-disable no-undef */
+const ws = new WebSocket('ws://localhost:8080');
+/* eslint-enable no-undef */
+let currentRoom = null;
+let currentRoomName = '';
+
+ws.onmessage = (event) => {
+  const data = JSON.parse(event.data);
+
+  switch (data.type) {
+    case 'roomCreated':
+      console.log(`Room created with ID: ${data.roomId}`);
+      break;
+
+    case 'roomJoined':
+      currentRoom = data.roomId;
+      currentRoomName = data.roomName || document.getElementById('room').value;
+
+      document.getElementById('chat').innerHTML =
+        `<h2>Messages in <span id="roomName">${currentRoomName}</span> chat:</h2>`;
+      data.messages.forEach((msg) => addMessage(msg));
+      enableChat();
+      document.getElementById('roomControls').style.display = 'block';
+      break;
+
+    case 'newMessage':
+      addMessage(data.message);
+      break;
+
+    case 'roomRenamed':
+      currentRoomName = data.newName;
+      document.getElementById('roomName').textContent = currentRoomName;
+      console.log(`Room renamed to: ${data.newName}`);
+      break;
+
+    case 'roomDeleted':
+      if (currentRoom === data.roomId) {
+        console.log('Room deleted, please join another room');
+        currentRoom = null;
+        currentRoomName = '';
+        document.getElementById('chat').innerHTML = '';
+        disableChat();
+        document.getElementById('roomControls').style.display = 'none';
+      }
+      break;
+  }
+};
+
+function setUsername() {
+  const username = document.getElementById('username').value;
+
+  if (!username) {
+    console.log('Please enter a username.');
+
+    return;
+  }
+  /* eslint-disable no-undef */
+  localStorage.setItem('username', username);
+  /* eslint-enable no-undef */
+
+  ws.send(JSON.stringify({ type: 'setUsername', username }));
+}
+
+function createRoom() {
+  const roomName = document.getElementById('room').value;
+
+  if (!roomName) {
+    console.log('Please enter a room name.');
+
+    return;
+  }
+  ws.send(JSON.stringify({ type: 'createRoom', roomName }));
+}
+
+function joinRoom() {
+  const roomId = document.getElementById('roomId').value;
+
+  if (!roomId) {
+    console.log('Please enter a valid room ID.');
+
+    return;
+  }
+  ws.send(JSON.stringify({ type: 'joinRoom', roomId }));
+}
+
+function sendMessage() {
+  const text = document.getElementById('message').value;
+
+  if (!text) {
+    console.log('Message cannot be empty');
+
+    return;
+  }
+  ws.send(JSON.stringify({ type: 'sendMessage', text }));
+  document.getElementById('message').value = '';
+}
+
+function addMessage(message) {
+  const chat = document.getElementById('chat');
+  const msgElem = document.createElement('div');
+
+  msgElem.classList.add('message');
+  msgElem.innerHTML = `<strong style="color: ${message.color}">${message.author}</strong>: <span>[${new Date(message.time).toLocaleTimeString()}]</span><p>${message.text}</p>`;
```

Comment:
Consider sanitizing the message content to prevent potential XSS attacks, especially if the message data comes from an untrusted source.